docs/Actions/inject-data.md

```diff
@@ -0,0 +1,274 @@
+---
+id: inject-data
+title: Inject Data
+sidebar_label: Inject Data
+---
+
+> Audience: [`Business Users`](/docs/audience#business-users)<br/>
+> Skill Prerequisites: `None`
+
+An action that lets you add data as tokens to the current execution context. Inject data can be used to load data that other actions other actions down the stack need.
+
+## `Typical Use Cases`
+
+- Generate a new token
+- Concatenate 2 tokens
+- Overwrite a token
+- Initialize variables with empty strings
+
+## `Don't use it to`
+
+- Execute a Variables token - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Apply tokens inside a text from different sources - use [Apply Tokens](/docs/Actions/apply-tokens.md) or [Execute Tokens](/docs/Actions/execute-tokens.md) actions instead
+- Create a token that is used in another execution context - store the data in a [hidden field](/docs/Actions/hidden-field.md), cookies or server session instead
+
+## `Related Actions`
+
+| Action Name | Description|
+|-------------|------------|
+| [Run SQL Query](/docs/Actions/run-sql-query.md) | Executes an SQL statement and captures the output, so that it can be used in the form using Inject Data.|
```

Comment:
Why do we mention "form" here? Is it specific to a form?
I still don't see a very strong relationship between these 2 actions. Perhaps if you reprhase it to say something like creating new tokens derived from data returned by the SQL query.

What about Execute Actions on Entity List? I remember we sometimes use inject data to create a token that gets appended with some text by each iteration.

Reply:
The last suggestion is a "hack" and we suggest using MyTokens instead, usually.